hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateReqHeader.java

```diff
@@ -47,6 +49,14 @@ public void populateReqHeader(Exchange exchange, @Headers Map<String, Object> hm
 		hm.put(MESSAGE_TYPE, msgType);
 		hm.put(SENDING_FACILITY, sendingFacility);
 		hm.put(Exchange.HTTP_RESPONSE_CODE, HttpStatus.OK_200);
+		
+		if(StringUtils.isNotEmpty(msgType) && msgType.equals(Util.MESSAGE_TYPE_PNP)) {
```

Comment:
Not significant but think you could use:
		Util.MESSAGE_TYPE_PNP.equals(msgType); 
or
		StringUtils.equals(Util.MESSAGE_TYPE_PNP, msgType);

to avoid NPE without doing the isNotEmpty check first. Also, just to note for these checks, that isNotBlank() is usually more what you want as it checks for empty strings also e.g. "   "

Reply:
done